packages/ui/src/views/repo/repo-commits/repo-commits-view.tsx

```diff
@@ -0,0 +1,105 @@
+import { NoData } from '@components/no-data'
+import { PaginationComponent } from '@components/pagination-component'
+import { SkeletonList } from '@components/skeleton-list'
+import { Spacer } from '@components/spacer'
+import { Text } from '@components/text'
+import { BranchSelectorListItem, SandboxLayout, TranslationStore } from '@views/index'
+
+import { BranchSelector } from '../components/branch-selector/branch-selector'
+import { CommitsList } from './components/commits-list'
+import { Filter } from './components/filter'
+import { TypesCommit } from './types'
+
+interface RepoCommitsViewProps {
+  isFetchingCommits: boolean
+  commitsList: TypesCommit[] | null | undefined
+  isFetchingBranches: boolean
+  branches?: {
+    name?: string
+    sha?: string
+  }[]
+  selectedBranch: BranchSelectorListItem
+  selectBranch: (branch: BranchSelectorListItem) => void
+  xNextPage: number
+  xPrevPage: number
+  page: number
+  setPage: (page: number) => void
+  sortOptions: {
+    name: string
+  }[]
+  sort: string | null
+  setSort: (sort: string) => void
+  query: string | null
+  setQuery: (query: string) => void
+  repoId: string
+  tagList: BranchSelectorListItem[]
+  spaceId: string
+  useTranslationStore: () => TranslationStore
+}
+
+export const RepoCommitsView = (props: RepoCommitsViewProps) => {
+  const { isFetchingCommits, isFetchingBranches, branches, selectedBranch, selectBranch } = props
+
+  return (
+    <SandboxLayout.Main hasHeader hasSubHeader hasLeftPanel>
+      <SandboxLayout.Content>
+        <Spacer size={10} />
+        <Text size={5} weight={'medium'}>
+          Commits
+        </Text>
+        <Spacer size={6} />
+        <div className="flex justify-between gap-5">
+          {!isFetchingBranches && branches && (
+            <BranchSelector
+              branchList={branches.map(item => ({
+                name: item.name || '',
+                sha: item.sha || ''
+              }))}
+              selectedBranch={selectedBranch}
+              tagList={props.tagList}
+              onSelectBranch={selectBranch}
+              repoId={props.repoId}
+              spaceId={props.spaceId}
+              useTranslationStore={props.useTranslationStore}
+            />
```

Comment:
maybe de-structure all props?

Reply:
We decided to do this on purpose. So that the dev knows they are coming from props and not a local variable. Some props that are reused may be de-structured.